sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -52,326 +51,383 @@ private Transforms() {
-                        break;
-                    case "Subtotal":
-                        extractedReceiptItem.setSubtotal(setFieldValue(fieldValue, readResults, includeTextDetails));
-                        break;
-                    case "Tax":
-                        extractedReceiptItem.setTax(setFieldValue(fieldValue, readResults, includeTextDetails));
-                        break;
-                    case "Tip":
-                        extractedReceiptItem.setTip(setFieldValue(fieldValue, readResults, includeTextDetails));
-                        break;
-                    case "Total":
-                        extractedReceiptItem.setTotal(setFieldValue(fieldValue, readResults, includeTextDetails));
-                        break;
-                    case "TransactionDate":
-                        extractedReceiptItem.setTransactionDate(setFieldValue(fieldValue, readResults, includeTextDetails));
-                        break;
-                    case "TransactionTime":
-                        extractedReceiptItem.setTransactionTime(setFieldValue(fieldValue, readResults, includeTextDetails));
-                        break;
-                    case "Items":
-                        extractedReceiptItem.setReceiptItems(toReceiptItems(fieldValue.getValueArray(), readResults, includeTextDetails));
-                        break;
-                    default:
-                        extractedFieldMap.putIfAbsent(key, setFieldValue(fieldValue, readResults, includeTextDetails));
-                        break;
+            extractedFormList.add(new RecognizedForm(extractedFieldMap, formType.get(), pageRange.get(),
+                new IterableStream<FormPage>(formPages.subList(pageRange.get().getStartPageNumber(), pageRange.get().getEndPageNumber()))));
+        }
+
+        // labeled
+        if (pageResults != null) {
+            for (int i = 0; i < pageResults.size(); i++) {
+                PageResult pageResultItem = pageResults.get(i);
+                Integer pageNumber = pageResultItem.getPage();
+
+                if (CoreUtils.isNullOrEmpty(documentResults)) {
+                    Integer clusterId = pageResultItem.getClusterId();
+                    if (clusterId != null) {
+                        formType.set(formType.get() + clusterId);
+                    }
+                    extractedFieldMap = getUnlabeledFieldMap(includeTextDetails, readResults, pageResultItem, pageNumber);
                 }
-            });
-            // adding other detected extra fields
-            extractedReceiptItem.setExtractedFields(extractedFieldMap);
-            extractedReceiptList.add(extractedReceiptItem);
+                extractedFormList.add(
+                    new RecognizedForm(extractedFieldMap, formType.get(),
+                        new PageRange(pageNumber, pageNumber),
+                        new IterableStream<FormPage>(Arrays.asList(formPages.get(pageNumber - 1)))));
```

Comment:
It may be more performant to use `Collections.singletonList` here, note that this will create an unmodifiable list so if this will be added onto that will throw an exception.